flink-runtime/src/main/java/org/apache/flink/runtime/io/network/ConnectionID.java

```diff
@@ -43,18 +46,36 @@ public class ConnectionID implements Serializable {
 
     private final int connectionIndex;
 
+    /**
+     * Resource ID can be known from shuffleDescriptor, to avoid rpc overhead,
+     * PartitionConnectionInfo will carry a connectionId with null resourceID.
```

Comment:
This is because `NettyShuffleDescriptor#fromProducerDescriptor` will create a `NetworkPartitionConnectionInfo`, which will carry a `ConnectionID`. If this connectionId carries a `ResourceID`, the RPC message will become larger.

BTW. Through offline discussion with @xintongsong , it is better to directly maintain address and connectionIndex in `NetworkPartitionConnectionInfo`. This can avoid the potentially null `ResourceID` and reduce the network transmission cost also.